### PR TITLE
Migration to add indices on foreign keys in new results table

### LINKED ISSUE #3549: Add indices on results table
### Background
The new results table will need indices for all the foreign keys we're using (test order, test event, and disease). We may also want to index on the results themselves, in the case of fetching all positive results.

### Action requested
Add indices on the results table for the following:
- test order
- test event
- disease

### Acceptance Criteria
Indices exist for the above 3 columns.

### PR DESCRIPTION
**Pull request by nickclyde**
# DATABASE PULL REQUEST

## Related Issue

- Resolves #3549

## Changes Proposed

- Adds indices on the foreign keys to diseases, test events and test orders on the new results table

## Additional Info

- [Some interesting reading](https://www.cybertec-postgresql.com/en/index-your-foreign-key/)

## Testing

- Run migration locally and rollback, ensure successful

## Screenshots

Before applying:
Screen Shot 2022-04-14 at 1 28 45 PM
Successful migration:
Screen Shot 2022-04-14 at 1 29 53 PM
After applying:
Screen Shot 2022-04-14 at 1 30 24 PM
Successful rollback:
Screen Shot 2022-04-14 at 1 31 57 PM

## Checklist for Primary Reviewer

- [x] Only database changes are included in this PR
- [x] Any new tables or columns that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
- [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views (including re-granting permission to the no-PHI user if need be)
- [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] Rollback has been verifed locally and in a deployed environment

### CODE REVIEW

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -3725,4 +3725,29 @@ databaseChangeLog:
...
-          ALTER TABLE ${database.defaultSchemaName}.result DROP COLUMN test_result;
\ No newline at end of file
+          ALTER TABLE ${database.defaultSchemaName}.result DROP COLUMN test_result;
+  - changeSet:
+      id: add-foreign-key-indices-to-result-table
+      author: nclyde@skylight.digital
+      comment: Add indices to result table to speed up queries
+      changes:
+        - tagDatabase:
+          tag: add-foreign-key-indices-to-result-table
```
Comment (zdeveloper) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:3735]: Tag is not being read correctly, I am assuming its not in the correct yaml place, btw also noticed the last migration had the same issue

**Review by zdeveloper — CHANGES_REQUESTED** [review]

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
Comment (nickclyde) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml]: Fixed!

**Review by emmastephenson — APPROVED** [review]
LGTM, thanks for adding these!

**Review by zdeveloper — APPROVED** [review]
LGTM! Thank you

**Pull request merged**